### DISCUSSION TITLE
The relationships of the models disappear

### DISCUSSION THREAD

**Discussion by ano101** [livewire/livewire#8185] [1. Help] [answered]
hello everyone. I have a parent component of the shopping cart. And there are child components of "products" in it. And when updating the parent component. all models in the child have relationships that disappear. I made a separate test property, and when mounting, I recorded a model with relationships, and when updating the parent component, it also lost relationships. Am I doing something wrong?

**Comment by ano101** [comment]
as I understand it, with Hydration, all models in the properties, even the one that is passed, they are serialized

**Comment by joshhanley** [comment, marked as answer]
@ano101 Livewire has troubles hydrating anything other than the models themselves. So it’s recommended that, if you need more that just the model, instead of adding a model to a public property that you use computer properties instead. See here for more information https://livewire.laravel.com/docs/properties#eloquent-constraints-arent-preserved-between-requests

**Comment by theticraft** [comment]
If I need to get the models in `mount()`, change and save them after at least one dehydrating - hydrating cycle: How will I do that?
Is a self written array the best option?
